### ISSUE TITLE
Arguments format  of `lingui add-locale` may need to be more flexible?

### ISSUE THREAD

**Issue by purepennons** [lingui/js-lingui#182]
In [`Intl`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl) (BCP 47), there are [some languages](https://www.iana.org/assignments/language-subtag-registry/language-subtag-registry) named like `zh-Hans-TW`, `zh-Hans-CN` . It makes more sense to name the locale folder same as Intl naming, or it may be some confusion when importing the message files into `language` prop of `I18nProvider`. 

https://github.com/lingui/js-lingui/blob/0d0aa92e334909a27ec0a38b2c99559e3c4558e5/packages/cli/src/api/formats/utils/locales.js#L16

```js
// locale/zh_TW/messages.js
module.exports = {...}

// usage
import zh_TW from './i18n/locale/zh_TW/messages'

const current_language = 'zh-Hant-TW'

// Need to rename.
// But maybe do not know that zh_TW need to rename to 'zh-Hant-TW' with some developers. 
<I18nProvider language={current_language} catalogs={{ 'zh-Hant-TW':  zh_TW}}>
    <App/>
</I18nProvider>

// some component used in App
// If dose not change zh_TW to zh-Hant-TW
// thow error -> RangeError: Invalid language tag: zh_TW
const C = () => <Trans><DateFormat value={lastLogin} /></Trans>  
```

BTW, there is some strange things that `zh` is a valid naming in `Intl` but `zh_TW` or `zh_CN` don't.

**Comment by tricoder42** [comment]
Hey @purepennons, sorry for lack of communication. Have you resolved this limitation in some other way or is it still issue? If so, I would leave it as open so I can fix it when I come across it.

**Comment by purepennons** [comment]
@tricoder42 
Thank you for your reply. This is still a issue. I just rename the keys before exporting/using them in my project.

**Comment by tricoder42** [comment]
Note to self: [bcp-47](https://www.npmjs.com/package/bcp-47) might be a good choice for validating locales instead of homemade regexp.

**Comment by tricoder42** [comment]
Released in [2.4.0](https://github.com/lingui/js-lingui/releases/tag/v2.4.0)
